src/debugger/Debugger.cpp

```diff
@@ -275,14 +275,19 @@ void Debugger::getBacktrace(ExecutionState* state, uint32_t minDepth, uint32_t m
     sendType(ESCARGOT_MESSAGE_BACKTRACE_END);
 }
 
-void Debugger::getScopeChain(ExecutionState* state)
+void Debugger::getScopeChain(ExecutionState* state, uint32_t scope)
 {
     const size_t maxMessageLength = ESCARGOT_DEBUGGER_MAX_MESSAGE_LENGTH - 1;
     uint8_t buffer[maxMessageLength];
     size_t nextScope = 0;
 
     LexicalEnvironment* lexEnv = state->lexicalEnvironment();
 
+    while (lexEnv && scope > 0) {
+        lexEnv = lexEnv->outerEnvironment();
+        scope--;
+    }
```

Comment:
I might misunderstand something, but you should go up in the stack, not in the lex env chain.

Reply:
+1